File: src/main/java/ar/edu/unq/spring/modelo/Item.java
package ar.edu.unq.spring.modelo;

import jakarta.persistence.*;
import lombok.*;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

@Entity
// @Cacheable
// @org.hibernate.annotations.Cache(usage = org.hibernate.annotations.CacheConcurrencyStrategy.READ_WRITE)
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    private int peso;

    @ManyToOne
    private Personaje owner;

    public Item(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

}

File: src/main/kotlin/ar/edu/unq/spring/modelo/Item.kt
package ar.edu.unq.spring.modelo

import javax.persistence.*
import java.util.Objects

@Entity
class Item() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var nombre: String? = null
    var peso: Int = 0

    @ManyToOne
    var owner: Personaje? = null

    constructor(nombre: String, peso: Int):this() {
        this.nombre = nombre
        this.peso = peso
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as Item?
        return id == item!!.id
    }

    override fun hashCode(): Int {
        return Objects.hash(id)
    }

    override fun toString(): String {
        return nombre!!
    }
}
